### PR TITLE
Display the fiat price on amounts and fees

### LINKED ISSUE #91: Display fiat price of L1 tokens
We should display the fiat price of L1 tokens retrieved by querying the Uniswap v3 smart contract.

### PR DESCRIPTION
**Pull request by amonsosanz**
Closes #91 

### What does this PR does?

This PR replaces the price oracle from uniswap v3 to v2 and implements the fiat display of amounts and fees, according to the currency selected by the user.

### CODE REVIEW

File: src/utils/amounts.ts [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+  const trimed =
+    decimals.length > TOKEN_DISPLAY_PRECISION
+      ? decimals.slice(0, TOKEN_DISPLAY_PRECISION)
+      : decimals;
+  return trimed === "" || trimed === "0" ? whole : `${whole}.${trimed}`;
+};
+
+export const formatFiatAmount = (value: BigNumber): string => {
+  const [whole, decimals = ""] = fiatBigNumberToString(value).split(".");
+  const trimed =
```
Comment (elias-garcia) [inline src/utils/amounts.ts:23]: It should be `trimmed`

File: src/utils/amounts.ts [hunk 2]
```diff
@@ -1,9 +1,49 @@
...
+
+interface Amount {
+  value: BigNumber;
+  precision: number;
+}
+
+export const multiplyAmounts = (a: Amount, b: Amount, outputPrecision: number): BigNumber => {
+  const result = formatUnits(a.value.mul(b.value), a.precision + b.precision);
+  const [whole, decimals = ""] = result.split(".");
+  const trimedDecimals =
```
Comment (elias-garcia) [inline src/utils/amounts.ts:46]: It should be `trimmedDecimals`

File: src/contexts/price-oracle.context.tsx [hunk 3]
```diff
@@ -31,57 +55,89 @@ const priceOracleContext = createContext<PriceOracleContext>({
...
+
+      if (fiatExchangeRate === undefined) {
         throw new Error("Fiat exchange rate not found");
       }
 
-      return usdPrice * fiatExchangeRate;
+      const price = multiplyAmounts(
+        { value: rate, precision: env.fiatExchangeRates.usdcToken.decimals },
+        {
+          value: fiatStringToBigNumber(fiatExchangeRate.toString()),
```
Comment (elias-garcia) [inline src/contexts/price-oracle.context.tsx:120]: This function seems to be only used here, can't we just use `parseUnits(fiatExchangeRate.toString(), PREFERRED_CURRENCY_ARITHMETIC_PRECISION);`.

File: src/utils/amounts.ts [hunk 4]
```diff
@@ -1,9 +1,49 @@
...
+};
+
+interface Amount {
+  value: BigNumber;
+  precision: number;
+}
+
+export const multiplyAmounts = (a: Amount, b: Amount, outputPrecision: number): BigNumber => {
+  const result = formatUnits(a.value.mul(b.value), a.precision + b.precision);
+  const [whole, decimals = ""] = result.split(".");
```
Comment (elias-garcia) [inline src/utils/amounts.ts:45]: Why do we need to split the `whole` part and the `decimals` to then do ``parseUnits(`${whole}.${trimedDecimals}`, outputPrecision)``?

Comment (elias-garcia) [reply src/utils/amounts.ts:45]: The `outputPrecision` is being used to later be able to retrieve the `fiatAmount` inside the `formatFiatAmount` function?

File: src/contexts/price-oracle.context.tsx [hunk 5]
```diff
@@ -1,28 +1,47 @@
...
+} from "src/types/contracts/uniswap-v2-router-02";
+import { UniswapV2Pair__factory } from "src/types/contracts/uniswap-v2-pair";
+import { FiatExchangeRates } from "src/domain";
+import { getFiatExchangeRates } from "src/adapters/fiat-exchange-rates-api";
+import { getCurrency } from "src/adapters/storage";
+import { Token, Chain } from "src/domain";
+import { getChainTokens, UNISWAP_V2_ROUTER_02_CONTRACT_ADDRESS } from "src/constants";
+
+const INIT_CODE_HASH = "0x96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f";
+const FACTORY_ADDRESS = "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f";
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/contexts/price-oracle.context.tsx]: Done

File: .env.integration [hunk 6]
```diff
@@ -1,7 +1,10 @@
+# Important: Inline comments are currently not supported in this env file
+
 # ETHEREUM
 REACT_APP_ETHEREUM_RPC_URL=http://localhost:8545
-REACT_APP_ETHEREUM_EXPLORER_URL=http://localhost:4000
+REACT_APP_ETHEREUM_EXPLORER_URL=https://goerli.etherscan.io
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply .env.integration]: AFAIK they don't deploy any L1 explorer in integration. This is a mere fake value so the env can be decoded. I've changed this to just `http://localhost`, removing the misleading `:4000`, which is the default port of the L2 explorer.

File: src/utils/amounts.ts (hunk 1 above)
Comment (amonsosanz) [reply src/utils/amounts.ts]: Fixed

File: src/utils/amounts.ts (hunk 2 above)
Comment (amonsosanz) [reply src/utils/amounts.ts]: Fixed

File: src/contexts/price-oracle.context.tsx (hunk 3 above)
Comment (amonsosanz) [reply src/contexts/price-oracle.context.tsx]: Agreed. The functions `fiatStringToBigNumber` and `fiatBigNumberToString` have been removed.

**Review by elias-garcia — APPROVED** [review]

**Pull request merged**